### PR TITLE
fix ShOperator. add shell option.

### PR DESCRIPTION
**Pull request by toyama0919**
Hello.

I want use login shell.
In addition, I want to choose shell environment variable because it is fixed in "/bin/sh".

Thanks.

### CODE REVIEW

**Comment by toyama0919** [comment]
This PR failed test?
My Local OSX is passed test.

**Comment by frsyuki** [comment]
Is difference of shell implementations significant? What kinds of shell do you use?
I think that workflow definition should be reproducible on many environments as much as possible. This change makes workflow more dependent on the execution environments. Alternative idea is to add `shell: /bin/sh` as an option.

**Comment by toyama0919** [comment]
Hmmm..

I want use server environments.
(rbenv, rvm, rails ,java environment... etc)
Then starting cost is very high.

My second idea, default_shell option.
What do you think?

```yaml
  sh>: echo hoge
  default_shell: zsh -l -c
```

**Comment by frsyuki** [comment]
Is `default_` prefix necessary? I think `shell` option is the most flexible:

```
_export:
  sh:
    shell: zsh -l
+step1:
  sh>: echo this uses zsh with -l option
```

The default `shell` is `/bin/sh`.

**Comment by toyama0919** [comment]
@frsyuki 
Ah, OK!
Thanks good advise.

**Comment by toyama0919** [comment]
@frsyuki 
>The default value of shell option is /bin/sh
fixed.

File: digdag-standards/src/main/java/io/digdag/standards/operator/ShOperatorFactory.java [hunk 1]
```diff
@@ -66,8 +70,13 @@ public TaskResult runTask()
             Config params = request.getConfig()
                 .mergeDefault(request.getConfig().getNestedOrGetEmpty("sh"));
 
+            String shell = params.get("shell", String.class, "/bin/sh -c");
             String command = params.get("_command", String.class);
-            ProcessBuilder pb = new ProcessBuilder("/bin/sh", "-c", command);
+
+            List<String> cmdline = new ArrayList<String>(Arrays.asList(shell.split(" ")));
```
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/ShOperatorFactory.java]: Splitting on whitespace seems a little brittle. How about making `shell` a list instead?

**Comment by toyama0919** [comment]
@danielnorberg
Thanks advise!

fixed bellow settings.
Please review.

```ruby
_export:
  sh:
    shell:
      - /bin/zsh
      - '-l'
      - '-c'
```

**Comment by frsyuki** [comment]
including -c is not good idea because it's assuming internal behavior of ShOperator

**Comment by frsyuki** [comment]
@danielnorberg I think it's better if sh operator can accept string as well as array because of #136 or #25 issue. Necessary update is to parse the input using a better tokenizer. Maybe that's a future improvement (because I can't come up with easy code to parse).

**Comment by frsyuki** [comment]
I added a commit and created #143. shell option is now more similar to shebang. So this also works:

```
+calling:
  sh>: p 1 + 1
  shell: [/usr/bin/env, ruby]
  #=> 2
```

**Comment by toyama0919** [comment]
@frsyuki 
Oh, thanks.
That is easy to understand.

**Pull request merged**